### ISSUE TITLE
Extract Variable breaks the React Component props and expressions

### ISSUE THREAD

**Issue by jackblackevo** [ThePrimeagen/refactoring.nvim#406]
Hi, I am using refactoring.nvim version 85cf9cb.
I noticed that Extract Variable breaks the React Component props and expressions.

Steps to reproduce:

1. Create a new `MyComponent.tsx` file:

```tsx
export const MyComponent = () => {
  return (
    <div>
      <h1>Hello World!</h1>
      <a download href="/hello.jpg">
        download
      </a>
    </div>
  );
};
```

2. Select lines 5-7 in visual mode.
3. Press `<leader>rv` to Extract Variable: `downloadLink`.

Result:

```tsx
export const MyComponent = () => {
  const downloadLink = <div>
      <h1>Hello World!</h1>
      <adownloadhref="/hello.jpg">
        download
      </a>
    </div>;
  return (
    < downloadLink />
  );
};
```

Expected:

```tsx
export const MyComponent = () => {
  const downloadLink = (
    <a download href="/hello.jpg">
      download
    </a>
  );

  return (
    <div>
      <h1>Hello World!</h1>
      {downloadLink}
    </div>
  );
};
```

Note: The `< downloadLink />` should be replaced with `{downloadLink}` to correctly render the extracted variable.

**Comment by mmirus** [comment]
Yep, I'm seeing this, as well. As shown above:

1. All spaces are stripped from the code that is extracted to the variable, breaking it.
2. `< variableName />` is inserted at the extraction site instead of `{variableName}`.

**Comment by TheLeoP** [comment]
The space related problem can be worked on, I'm on it. But the original problem is because of how treesitter works (you probably selected not only the tag, but the whitespace arround it), check #404 For a better explanation of why this happens

**Comment by mmirus** [comment]
@TheLeoP Thanks for working on the space problem! Can you clarify what you mean by the "original problem"? Do you mean how the refactor results in `< varName />` instead of `{varName}`?

**Comment by TheLeoP** [comment]
@mmirus oh, I didn't realize that the syntax was wrong, I'll work on that too, should be a simple fix. I was referring to the fact that the variable extracted selected a bigger region than the one you intended to extract

**Comment by mmirus** [comment]
Thank you!

> I was referring to the fact that the extracted variable selected a bigger region than the one you intended to extract

I didn't actually run into this problem, but I think @jackblackevo did. I think the difference is that I selected only the tag on lines 5-7 (`vat`) before refactoring, while he most likely selected the full lines.

**Comment by jackblackevo** [comment]
> > I was referring to the fact that the extracted variable selected a bigger region than the one you intended to extract
> 
> I didn't actually run into this problem, but I think @jackblackevo did. I think the difference is that I selected only the tag on lines 5-7 (`vat`) before refactoring, while he most likely selected the full lines.

Yes, I used the uppercase `V` to select lines 5-7.

**Comment by TheLeoP** [comment]
This should be closed by #412.

> I found that using the uppercase V to select lines 5-7 is not an issue for Extract Function:

@jackblackevo. Yes, that's because extract function works on statements (where each statement is one node, so it works with multiple nodes), while extract variable work on a expression (only one node). So, if you select multiple lines for both, extract function will extract all the statements contained in your selection, while extract variable will select the smallest node that contains all of your selection.
